pkg/workloadmeta/collectors/podman/podman.go

```diff
@@ -157,9 +158,42 @@ func (c *collector) expiredEvents() []workloadmeta.CollectorEvent {
 func networkIPs(container *podman.Container) map[string]string {
 	res := make(map[string]string)
 
-	if len(container.State.NetworkStatus) > 0 {
-		if len(container.State.NetworkStatus[0].IPs) > 0 {
-			res["podman"] = container.State.NetworkStatus[0].IPs[0].Address.IP.String()
+	// container.Config.Networks contains only the networks specified at container creation time
+	// and not the ones attached afterwards with `podman network attach`
+	// They appear in the order in which they were specified in the `podman run --net=…` command
+	networkNames := make([]string, len(container.Config.Networks))
+	copy(networkNames, container.Config.Networks)
+	sort.Strings(networkNames)
+
+	if len(container.Config.Networks) != len(container.State.NetworkStatus) {
+		var containerID string
+		if len(container.Config.ID) >= 12 {
+			containerID = container.Config.ID[:12]
+		} else {
+			containerID = container.Config.ID
+		}
+		log.Warnf("podman container %s %s has now a number of networks different from what it was at creation time. This can be due to the use of `podman network attach`/`podman network detach`. This may confuse the agent.", containerID, container.Config.Name)
```

Comment:
I'm not sure if it would be better to just return an empty mapping in this case, because we know that there are no guarantees that the mapping calculated in the lines below is going to be correct.

Reply:
Done.